File: java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

import static java.lang.Math.toIntExact;

public class Hamming {
    static int compute(String s1, String s2) {
        if (s1.length() != s2.length()) {
            throw new IllegalArgumentException("string length not equal.");
        }

        return toIntExact(IntStream.range(0, s1.length())
                .filter(i -> s1.charAt(i) != s2.charAt(i)).count());
    }
}

File: scala/hamming/src/main/scala/Hamming.scala
/**
  *
  * @author Cedric Zhuang
  *
  */
object Hamming {
  def compute(s1: String, s2: String): Option[Int] = {
    if (s1.length != s2.length) return None
    Some(s1.zip(s2).count { case (a, b) => a != b })
  }

}
